ffi/src/lib.rs

```diff
@@ -0,0 +1,187 @@
+//! A basic foreign function interface (FFI) for "only" reading information
+//! of [Windows Installer](https://en.wikipedia.org/wiki/Windows_Installer)
+//! (MSI) files, built on top of the [msi](https://crates.io/crates/msi) crate.
+
+#![warn(missing_docs)]
+
+use chrono::prelude::{DateTime, Utc};
+use msi::{Package, Select};
+use safer_ffi::prelude::*;
+use std::io;
+use std::path::Path;
+
+// ========================================================================= //
+
+/// Information about an MSI file.
+#[derive_ReprC]
+#[repr(C)]
+pub struct MsiInformation {
+    /// Architecture of the MSI installer.
+    arch: repr_c::String,
+    /// Author of the MSI file.
+    author: repr_c::String,
+    /// Comments about the MSI file.
+    comments: repr_c::String,
+    /// The application that created the MSI file.
+    creating_application: repr_c::String,
+    /// The creation time in RFC2822 format.
+    creation_time: repr_c::String,
+    /// The BCP-47 codes of the languages in the MSI file.
+    languages: repr_c::Vec<repr_c::String>,
+    /// The subject of the MSI file.
+    subject: repr_c::String,
+    /// The title of the MSI file.
+    title: repr_c::String,
+    /// The UUID of the MSI file.
+    uuid: repr_c::String,
+    /// Word count
+    word_count: i32,
+
+    /// Indicates whether the MSI file has a digital signature.
+    has_digital_signature: bool,
+
+    /// The names of the tables in the MSI database.
+    table_names: repr_c::Vec<repr_c::String>,
+}
+
+// ========================================================================= //
+
+/// Gets the information of an MSI file at the given path.
+#[ffi_export]
+fn get_information(path: char_p::Ref<'_>) -> MsiInformation {
+    let file_handle = std::fs::File::open(Path::new(path.to_str())).unwrap();
+    let package = Package::open(file_handle).unwrap();
```

Comment:
Note that this will panic (and presumably crash your C# program) if the MSI file is invalid (or on the previous line, if the file can't be read), which doesn't seem ideal.  Would it be possible to return some kind of error value, or even just a null value, when that happens?